server/job_service.go

```diff
@@ -169,6 +169,39 @@ func (p *Plugin) newJobService(serviceURL string) (*jobService, error) {
 		return nil, fmt.Errorf("failed to create client: %w", err)
 	}
 
+	err = client.Login(cfg.ClientID, cfg.AuthKey)
+	if err == nil {
+		return &jobService{
+			ctx:    p,
+			client: client,
+		}, nil
+	}
+
+	// If login fails we attempt to re-register once as the jobs instance may
+	// have restarted, potentially losing stored credentials.
+	p.LogError("failed to login to job service", "err", err.Error())
+	p.LogDebug("attempting to re-register the job service client")
+
+	mutex, err := cluster.NewMutex(p.API, "job_service_registration")
```

Comment:
Just curious why we use a cluster mutex here, and not above?

Reply:
It's used above as well, just buried in `getJobServiceClientConfig`

Reply:
Ah. Would it be too much to consolidate both of these, so one of them doesn't get accidentally removed or whatnot? Up to you.

Reply:
Maybe we could call it just once at the beginning. The down side is that all other instances will be blocked on this for potentially some time as there are requests being made but I guess during init that would be acceptable. I'll take another look.

Reply:
Thanks @cpoile , looks cleaner now and works as expected :+1: